src/brod_producer.erl

```diff
@@ -383,6 +384,9 @@ handle_cast(_Cast, #state{} = State) ->
 code_change(_OldVsn, #state{} = State, _Extra) ->
   {ok, State}.
 
+terminate(shutdown, #state{client_pid = ClientPid, topic = Topic, partition = Partition}) ->
+  brod_client:deregister_producer(ClientPid, Topic, Partition),
```

Comment:
please check terminate reason like what's done in `brod_consumer:terminate`

Reply:
happy to; thanks for the feedback!